File: app/src/main/java/no/nav/tilbakemeldingsmottak/rest/ros/validation/SendRosValidator.java
package no.nav.tilbakemeldingsmottak.rest.ros.validation;

import no.nav.tilbakemeldingsmottak.model.SendRosRequest;
import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator;
import org.springframework.stereotype.Component;

import static no.nav.tilbakemeldingsmottak.model.SendRosRequest.HvemRoses;

@Component
public class SendRosValidator extends RequestValidator {

    public void validateRequest(SendRosRequest request) {
        isNotNull(request.getHvemRoses(), "hvemRoses");
        if (HvemRoses.NAV_KONTOR.equals(request.getHvemRoses())) {
            hasText(request.getNavKontor(), "navKontor", " dersom hvemRoses=NAV_KONTOR");
        }
        hasText(request.getMelding(), "melding");
    }

}

File: app/src/main/kotlin/no/nav/tilbakemeldingsmottak/rest/ros/validation/SendRosValidator.kt
package no.nav.tilbakemeldingsmottak.rest.ros.validation

import no.nav.tilbakemeldingsmottak.model.SendRosRequest
import no.nav.tilbakemeldingsmottak.model.SendRosRequestHvemRoses
import no.nav.tilbakemeldingsmottak.rest.common.validation.RequestValidator
import org.springframework.stereotype.Component

@Component
class SendRosValidator : RequestValidator() {
    fun validateRequest(request: SendRosRequest) {
        isNotNull(request.hvemRoses, "hvemRoses")
        if (SendRosRequestHvemRoses.NAV_KONTOR == request.hvemRoses) {
            hasText(request.navKontor, "navKontor", " dersom hvemRoses=NAV_KONTOR")
        }
        hasText(request.melding, "melding")
    }
}
